File: dsa-pros/leetcode/stack-queue/javascript/cynthiaNwakaeme/week15/simplifyPath.js
/**
 * @param {string} path
 * @return {string}
 */

const simplifyPath = (path) => {
  const stack = [];
  const directories = path.split("/");

  for (const str of directories) {
    if (str === "." || !str) continue;
    else if (str === "..") {
      if (stack.length > 0) stack.pop();
    } else stack.push(str);
  }

  return `/${stack.join("/")}`;
}

console.log(simplifyPath("/../"));
// console.log(simplifyPath("/home/"));
// console.log(simplifyPath("/home//foo/"));

File: dsa-pros/leetcode/stack-queue/java/motunrayoIlawole/simplify-path.java
import java.util.Stack;

class Solution {
    public String simplifyPath(String path) {

        Stack<String> pathStack = new Stack<>();

        String[] pathLevels = path.split("/");

        for (int i = 0; i < pathLevels.length; i++) {
            if (pathLevels[i].equals("..") && !pathStack.isEmpty()) {
                pathStack.pop();
            } else if (!pathLevels[i].equals("..") && !pathLevels[i].equals(".") && !pathLevels[i].equals("")) {
                pathStack.push(pathLevels[i]);
            }
        }

        return "/" + String.join("/", pathStack);
    }
}